users/views.py

```diff
@@ -89,6 +92,26 @@ class UserCreate(BSModalCreateView):
     template_name = "modals/create_and_update_user.html"
     form_class = CustomUserCreationForm
 
+    def _send_create_user_email(self, user: CustomUser, password: str) -> None:
+        current_site = get_current_site(self.request)
+        message = render_create_user_email(user, current_site.domain, password)
+        send_email(mail_subject="Your Sheet Storm account has been created.", message=message, addressee=user.email)
+
+    def _create_user_with_random_password(self, form: CustomUserCreationForm) -> CustomUser:
+        user_random_password = generate_random_string_from_letters_and_digits(
+            UserConstants.RANDOM_PASSWORD_LENGTH.value
+        )
+        user = form.save(commit=False)
+        user.set_password(user_random_password)
+        user.is_active = True
+        user.need_change_password = True
+        user.save()
+        user_preferences = CustomUserPreferences(user=user)
+        user_preferences.full_clean()
+        user_preferences.save()
```

Comment:
It will be soon done as a signal and you will delete it probably before you merge your PR, but until this happen please also move this code to separate method

Reply:
Done.